app/services/top_news_service.rb

```diff
@@ -0,0 +1,18 @@
+# Returns the top news stories from 1-n feed sources as a Hash.
+class TopNewsService
+  include Limitable
+
+  def initialize(limit:)
+    @limit = limit
+
+    validate_limit!
+  end
+
+  def execute
+    # TODO: Add other news feed sources as needed and refactor
+    # the limit logic accordingly (e.g. split 1 + n feed source
+    # results according to limit).
+    ids = Feeds::HackerNews::TopStories.new(limit: limit).execute
+    Feeds::HackerNews::StoriesWithDetail.new(ids).execute
+  end
```

Comment:
Our TopNewsService service. This service retrieves the news feeds we want to consume from all our news sources to be placed in our feeds table.